Mage.Sets/src/mage/cards/s/SlicerHiredMuscle.java

```diff
@@ -0,0 +1,124 @@
+
+        // Double strike
+        this.addAbility(DoubleStrikeAbility.getInstance());
+
+        // Haste
+        this.addAbility(HasteAbility.getInstance());
+
+        // At the beginning of each opponent's upkeep, you may have that player gain
+        // control of Slicer until end of turn. If you do, untap Slicer, goad it, and it
+        // can't be sacrificed this turn. If you don't, convert it.
+        this.addAbility(new BeginningOfUpkeepTriggeredAbility(Zone.BATTLEFIELD, new SlicerHiredMuscleUpkeepEffect(),
+                TargetController.OPPONENT, false, true));
+
+    }
+
+    private SlicerHiredMuscle(final SlicerHiredMuscle card) {
+        super(card);
+    }
+
+    @Override
+    public SlicerHiredMuscle copy() {
+        return new SlicerHiredMuscle(this);
+    }
+}
+
+class SlicerHiredMuscleUpkeepEffect extends OneShotEffect {
+
+    public SlicerHiredMuscleUpkeepEffect() {
+        super(Outcome.GainControl);
+        staticText = "you may have that player gain control of {this} until end of turn. If you do, untap {this}, goad it, and it can't be sacrificed this turn. If you don't, convert it.";
+    }
+
+    public SlicerHiredMuscleUpkeepEffect(final SlicerHiredMuscleUpkeepEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public SlicerHiredMuscleUpkeepEffect copy() {
+        return new SlicerHiredMuscleUpkeepEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+
+        Permanent sourcePermanent = game.getPermanent(source.getSourceId());
+        Player player = game.getPlayer(sourcePermanent.getOwnerId());
+        Player newController = game.getPlayer(getTargetPointer().getFirst(game, source));
+
+        if (player == null || newController == null || sourcePermanent == null) {
+            return false;
+        }
+
+        if (player.chooseUse(this.getOutcome(), source.getRule(), source, game)) {
+
+            // Gain control
+            ContinuousEffect gainControllEffect = new GainControlTargetEffect(Duration.EndOfTurn, false,
+                    newController.getId());
+            gainControllEffect.setTargetPointer(new FixedTarget(sourcePermanent.getId(), game));
+            gainControllEffect.setText("and gains control of it");
+
+            // Untap
+            sourcePermanent.untap(game);
+
+            // Goad
+            game.addEffect(new GoadTargetEffect()
+                    .setDuration(Duration.EndOfTurn)
+                    .setTargetPointer(new FixedTarget(sourcePermanent, game)),
+                    source);
+
+            // Can't be sacrificed
+            game.addEffect(new GainAbilityTargetEffect(
+                    new SimpleStaticAbility(
+                            new CantBeSacrificedSourceEffect().setText("This creature can't be sacrificed")),
+                    Duration.EndOfTurn).setTargetPointer(new FixedTarget(sourcePermanent, game)), source);
+
+            game.addEffect(gainControllEffect, source);
```

Comment:
Must gain control first and make other actions after it due rules text order. 

I think opponent's triggers must trigger on untap or goad. Example: opponent with [[Wake Thrasher]] - it must be boostered on upkeep with new control.

Reply:
If not help with boost then try to call game.getstate.processactions after gain control.

Reply:
game.getstate.processactions did the job